src/snmalloc/mem/remotecache.h

```diff
@@ -12,6 +12,164 @@
 
 namespace snmalloc
 {
+  /**
+   * Stores the remote deallocation to batch them before sending
+   */
+  template<typename Config, size_t RINGS>
+  class RemoteDeallocCacheBatching
+  {
+    static_assert(RINGS > 0);
+
+    std::array<freelist::Builder<false, true>, RINGS> open_builder;
+    std::array<address_t, RINGS> open_meta = {0};
+
+    SNMALLOC_FAST_PATH size_t
+    ring_set(typename Config::PagemapEntry::SlabMetadata* meta)
+    {
+      // See https://github.com/skeeto/hash-prospector for choice of constant
+      return ((meta->as_key_tweak() * 0x7EFB352D) >> 16) &
+        bits::mask_bits(DEALLOC_BATCH_RING_SET_BITS);
+    }
+
+    template<typename Forward>
```

Comment:
Perhaps explain use of `Forward`?  Basically continuation?

Reply:
Comment added to "Initial dealloc-side batching machinery"